Project_JAVA (Thymeleaf) PAIVA ALEXY/src/main/java/com/myaudiolibary/web/controllerThym/AudioController.java

```diff
@@ -0,0 +1,132 @@
+package com.myaudiolibary.web.controllerThym;
+
+import com.myaudiolibary.web.exception.ConflictException;
+import com.myaudiolibary.web.model.Album;
+import com.myaudiolibary.web.model.Artist;
+import com.myaudiolibary.web.repository.AlbumRepository;
+import com.myaudiolibary.web.repository.ArtistRepository;
+import com.myaudiolibary.web.service.ArtistService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Sort;
+import org.springframework.http.MediaType;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.ModelMap;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.servlet.mvc.support.RedirectAttributes;
+import org.springframework.web.servlet.view.RedirectView;
+
+import javax.persistence.EntityNotFoundException;
+import java.util.Optional;
+
+@Controller
+@RequestMapping(value = "thym/artists")
+public class AudioController {
+
+    @Autowired
+    private ArtistRepository artistRepository;
+
+    @Autowired
+    private ArtistService artistService;
+
+
+    @GetMapping(value = "/{id}")
+    public String id(final ModelMap model, @PathVariable Long id) {
+        if (!artistRepository.existsById(id)) {
+            throw new EntityNotFoundException("L'identifiant " + id + " n'a pas été trouvé !");
+        } else {
+
+            Optional<Artist> ArtistOptional = artistRepository.findById(id);

```

Comment:
Utilise l'Optional directement pour éviter de faire deux requêtes BDD: 
```suggestion
        Optional<Artist> ArtistOptional = artistRepository.findById(id);
        if (!ArtistOptional.isPresent()) {
            throw new EntityNotFoundException("L'identifiant " + id + " n'a pas été trouvé !");
        } else {
```